cpp/biconnected_components_module/biconnected_components_test.cpp

```diff
@@ -0,0 +1,284 @@
+#include <chrono>
+#include <random>
+#include <set>
+#include <vector>
+
+#include <gtest/gtest.h>
+#include <mg_graph.hpp>
+
+#include "algorithm/biconnected_components.hpp"
+
+/// This class is threadsafe
+class Timer {
+ public:
+  Timer() : start_time_(std::chrono::steady_clock::now()) {}
+
+  template <typename TDuration = std::chrono::duration<double>>
+  TDuration Elapsed() const {
+    return std::chrono::duration_cast<TDuration>(std::chrono::steady_clock::now() - start_time_);
+  }
+
+ private:
+  std::chrono::steady_clock::time_point start_time_;
+};
+
+/// Builds the graph from a given number of nodes and a list of edges.
+/// Nodes should be 0-indexed and each edge should be provided in both
+/// directions.
+std::unique_ptr<mg_graph::Graph<>> BuildGraph(std::uint64_t nodes,
+                                              std::vector<std::pair<std::uint64_t, std::uint64_t>> edges) {
+  auto G = std::make_unique<mg_graph::Graph<>>();
+  for (std::uint64_t i = 0; i < nodes; ++i) G->CreateNode(i);
+  for (auto &p : edges) G->CreateEdge(p.first, p.second);
```

Comment:
`for (const auto [from, to] : edges)` or whatever name fits the best here.
Same in the other places, better than using those weird `first` and `second` names from pair. :smile: